include/swift/AST/AutoDiff.h

```diff
@@ -113,6 +113,13 @@ struct AutoDiffConfig {
   SWIFT_DEBUG_DUMP;
 };
 
+/// A semantic function result type: either a formal function result type or
+/// an `inout` parameter. Used in derivative function type calculation.
+struct SemanticFunctionResultType {
```

Comment:
Should make this name be more specific to AD, e.g. `AutoDiffSemanticFunctionResultType`.

Reply:
Good point. Done.